server/config.go

```diff
@@ -229,3 +234,70 @@ func RunSetPerm(ctx *Context, keyPrefix, configFileName string) {
 func RunSetZone(ctx *Context, keyPrefix, configFileName string) {
 	runSetConfig(ctx, zonePathPrefix, keyPrefix, configFileName)
 }
+
+// GetConfig retrieves the configuration for the specified key. If the
+// key is empty, all configurations are returned. Otherwise, the
+// leading "/" path delimiter is stripped and the configuration
+// matching the remainder is retrieved. Note that this will retrieve
+// the default config if "key" is equal to "/", and will list all
+// configs if "key" is equal to "". The body result contains a listing
+// of keys and retrieval of a config. The output format is determined
+// by the request header.
+func GetConfig(db *client.KV, configPrefix proto.Key, config gogoproto.Message, path string, r *http.Request) (body []byte, contentType string, err error) {
```

Comment:
Go style doesn't have an explicit line length limit, but I like to stay around 100 characters to make side-by-side diffs more readable. I'd probably put a newline before the `path string` parameter.